### PR TITLE
PIMOB-987: Add event for start of payment journey

### PR DESCRIPTION
**Pull request by alex-ioja-yang-cko**
## Proposed changes

[Jira task](https://checkout.atlassian.net/browse/PIMOB-987)

🟢 Add event to track start of payment journey
🔴 Did not add unit tests as they would require opening up and injecting a logger inside the builder. That would very much invalidate the purpose of the builder to be responsible for creating everything
🟡 We may be able to test if we wanted the builder to have state and limited the access for consumer. See code example
```swift
// not testable as we cannot inject state inside generateContent
Builder.generateContent()

// testable as we can use internal access control to override properties that will be used
Builder().generateContent()
```

## Further comments

Since the changes have been so insignificant, have allowed some SwiftLint fixes to blend in. Must have been introduced in the PR between starting the SwiftLint branch and its merging into main.

### CODE REVIEW

File: Source/Core/Logging/FramesLogEvent.swift [hunk 1]
```diff
@@ -49,8 +49,8 @@ enum FramesLogEvent: Equatable, PropertyProviding {
 
     private var typeIdentifierSuffix: String {
         switch self {
-        case .checkoutAPIClientInitialised:
-            return "checkout_api_client_initialised"
+        case .paymentJourneyStart:
+            return "payment_journey_start"
```
Comment (alex-ioja-yang-cko) [inline Source/Core/Logging/FramesLogEvent.swift:53]: @deepesh-vasthimal-cko & @patrick-hoban-cko follow up on the Slack discussion, let me know if I was convincing on the need to separate this event name from syntactically getting tied to the VC init and whether you have alternative names for event.

File: Source/UI/NewUI/PaymentForm/Factory/PaymentFormFactory.swift [hunk 2]
```diff
@@ -13,6 +13,7 @@ public struct PaymentFormFactory {
     // Ensure a consistent identifier is used for the monitoring of a journey
     Self.sessionCorrelationID = UUID().uuidString
     let logger = FramesEventLogger(environment: configuration.environment, getCorrelationID: { Self.sessionCorrelationID })
+    logger.log(.paymentJourneyStart(environment: configuration.environment))
```
Comment (alex-ioja-yang-cko) [inline Source/UI/NewUI/PaymentForm/Factory/PaymentFormFactory.swift:16]: @ehab-al-cko , @deepesh-vasthimal-cko see PR description for why this is not testable in current call site syntax. Let me know if you want to move from
```swift
PaymentFormFactory. buildViewController

// TO

PaymentFormFactory(). buildViewController
```

that apparent minor change makes all the difference :)

File: Tests/Mocks/MockCardValidator.swift [hunk 3]
```diff
@@ -23,11 +23,17 @@ final class MockCardValidator: CardValidating {
...
 
     var validateCVVToReturn: ValidationResult<ValidationError.CVV> = .success
+    var expectedMaxLenghtCVV = 4
+    var maxLenghtCVVCalledWith = [Card.Scheme]()
     private(set) var validateCVVCalledWith: (cvv: String, cardScheme: Card.Scheme)?
     func validate(cvv: String, cardScheme: Card.Scheme) -> ValidationResult<ValidationError.CVV> {
         validateCVVCalledWith = (cvv, cardScheme)
         return validateCVVToReturn
     }
+    func maxLenghtCVV(for scheme: Card.Scheme) -> Int {
```
Comment (patrick-hoban-cko) [inline Tests/Mocks/MockCardValidator.swift:33]: Sorry to be picky @alex-ioja-yang-cko, but there are a number of places in these fields where Length is spelt incorrectly as _Lenght_

Could we update that please?

File: Source/Core/Logging/FramesLogEvent.swift (hunk 1 above)
Comment (patrick-hoban-cko) [reply Source/Core/Logging/FramesLogEvent.swift:53]: I'll let @deepesh-vasthimal-cko give his steer.

File: Tests/Mocks/MockCardValidator.swift (hunk 3 above)
Comment (alex-ioja-yang-cko) [reply Tests/Mocks/MockCardValidator.swift:33]: waaaait, I'm working on different task on wrong branch, this should not be here 🙈. But will fix that in correct place once I steer it back 👍

Comment (alex-ioja-yang-cko) [reply Tests/Mocks/MockCardValidator.swift]: ✅ please look again, this time its just what's supposed to be here

File: Source/Core/Logging/FramesLogEvent.swift (hunk 1 above)
Comment (patrick-hoban-cko) [reply Source/Core/Logging/FramesLogEvent.swift:53]: @alex-ioja-yang-cko let's go with _payment_form_initialised_

That means we have more choice later on if there's another way to begin the payment journey in the future (for example, what if we have some other sort of integration option of the SDKs, I'd want to be distinct about which event I'm actually looking at).

**Review by patrick-hoban-cko — APPROVED** [review]

**Review by ehab-al-cko — APPROVED** [review]

**Pull request merged**